music-journal/components/Memory.js

```diff
@@ -1,12 +1,46 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import withAuth from '../utils/withAuth';
 import AddMemory from '../components/AddMemory'
 import ListMemories from '../components/ListMemories'
+import { addMemory, getMemories } from '../utils/fetcher';
+
 const Memory = (props) => {
-    // Set state here and pass it down so that can automatically refresh the list on successful submission?
+    const [memories, setMemories] = useState()
+    const [text, setText] = useState('')
+    const [song, setSong] = useState('')
+
+    useEffect(async () => {
+        try {
+            const response = await getMemories(props.token)
+            setMemories(response)
+        } catch (error) {
+            console.log(error)
+        }
+    }, []);
+
+    const handleSubmit = async (event) => {
+        event.preventDefault()
+
+        try {
+            const newMemory = await addMemory({ song, text }, props.token)
+            setSong('')
+            setText('')
+            setMemories({ ...memories, ...newMemory })

```

Comment:
so this essentially appends the new memory to the existing list of memories, then sets the memories to be the new list?

Reply:
Yep, that's it